Add CROSS_BOOTSTRAP_JULIA option for building the sysimg with another Julia

xal-0 opened the pull request:
Adds a makefile variable CALLED `CROSS_BOOTSTRAP_JULIA` that controls the Julia
executable and `sysbase.so` used to build `sys.so`.  This is really useful for
building a ThreadSanitizer-enabled version of Julia in a reasonable amount of
time.  It is used like this:

```
make O=bootstrap julia-src-release julia-sysbase-release
make CROSS_BOOTSTRAP_JULIA=bootstrap/usr/ julia-src-release julia-sysimg-release
```

The `--target-sanitize-thread` option set in `Make.inc` is implemented in
another pull request, but shouldn't affect CI yet; the bootstrap version of
Julia will be built but won't be used.

Also brings `COMPILER_SRCS` up to date.

xal-0 committed:
Update COMPILER_SRCS

xal-0 committed:
Add CROSS_BOOTSTRAP_JULIA option

Keno commented:
Can we use this to fix https://github.com/JuliaLang/julia/issues/58068 (with an option to do a full debug bootstrap as right now)?

xal-0 committed:
Rename --target-sanitize-thread to --target-sanitize=thread

xal-0 committed:
Disable tsan bug reports while bootstrapping

xal-0 committed:
Split CROSS_BOOTSTRAP_JULIA and CROSS_BOOTSTRAP_SYSBASE

This changes makes it possible to create debug builds of Julia faster, by
pointing CROSS_BOOTSTRAP_JULIA at ./usr/bin/julia and CROSS_BOOTSTRAP_SYSBASE at
./usr/lib/julia/sysbase.so.

xal-0 committed:
Add devdocs for cross-bootstrapping a debug build of Julia

xal-0 committed:
Replace tortured sentence in build.md

xal-0 committed:
Fix CROSS_BOOTSTRAP_SYSBASE typo in Make.user.tsan

xal-0 committed:
Revert "Update COMPILER_SRCS" to fix merge conflict (sniped by Claude)

This reverts commit a3b143264937eb2438c07a16e4fd4efef3b0b276.
